polaris/mixins/_checksum.py

```diff
@@ -66,6 +64,9 @@ def verify_checksum(self, md5sum: str | None = None):
         self.md5sum = self._compute_checksum()
 
         if self.md5sum != md5sum:
+            # Imported here to prevent circular import
+            from polaris.utils.errors import PolarisChecksumError
```

Comment:
> (Ironically, one of the rare instance where having barrel files all over the place revealed a problem with the module structure, rather than caused one...)

I'm not sure what _barrel files_ are, but I would be interested to learn more about "module structure". We are more frequently running into circular imports in Polaris, so there is something valuable to learn there I think! 😄

Reply:
On this issue - I removed the formatting from the error messages. In addition to the module structure, I feel like improving our logging and error system are two other topics that I know little about, but that seem inevitably important to "production-ready" software and that I would thus love to learn more about!

Reply:
A barrel file is a name (that originated from the JavaScript/TypeScript world) used to describe a file that only imports and then reexports other elements. This is what the `__init__.py` files do in this package.

The module structure is basically how you split the code of the package. Ideally, it would look like a pyramid of successive layers, where higher-level layers, coarser layers of functionality call lower-level, finer grained layers of functionality. More formally, your dependency graph needs to be a DAG to avoid issues, and also to simplify the mental model of how pieces work together.

If you find yourself fighting circular imports, that means that you don't have a DAG and some lower level layer is calling on a higher level layer.